### PR TITLE
feat: extract variables from scoped css

### PR DESCRIPTION
**Pull request by KnisterPeter**
ISSUES CLOSED: #11

### CODE REVIEW

File: src/css.js [hunk 1]
```diff
@@ -4,37 +4,121 @@ const decamelize = require('decamelize');
...
+	if (cssVariables) {
+		resolveScopeVariables(cssom, vars);
+	}
+	const hash = createHash(css.stringify(cssom));
+	const classNames = getClassNames(`${scope.ns}-${hash}`, cssom);
+
+	return [classNames, vars, css.stringify(cssom)];
+}
+
+function createHash(input) {
```
Comment (KnisterPeter) [inline src/css.js]: Do you have one? This is the (fast) java hash-code implementation.
It create fast and reliable hash codes without too much collisions.

File: src/index.js [hunk 2]
```diff
@@ -27,30 +26,30 @@ function logResult(promise) {
...
-			.then(([components, filepaths, statics]) =>
-				renderPages(filepaths, dest, {components, vars, statics, disableValidation, scopedCss})
+				.then(statics => [templates, filepaths, statics]))
+			.then(([templates, filepaths, statics]) =>
+				renderPages(filepaths, dest, {templates, vars, statics, disableValidation, cssVariables})
 					.then(pageStylesheets => {
 						if (scopedCss) {
-							return sander.writeFile(scopedCss, combineCss(components, pageStylesheets));
+							return sander.writeFile(scopedCss,
+								Array.from(new Set(pageStylesheets.join('\n').split('}').map(rule => rule.trim() + '}'))).join('\n'));
```
Comment (marionebl) [inline src/index.js]: Thats very very terse, please split into at least two lines

File: src/css.js (hunk 1 above)
Comment (marionebl) [inline src/css.js]: Perhaps a comment explaining the source for this / what is happening exactly would be in order. I personally have trouble grasping it to full extent

File: src/css.js [hunk 3]
```diff
@@ -4,37 +4,121 @@ const decamelize = require('decamelize');
...
+ * @param {string[]} scopedCss A list of page CSS
+ * @returns {string} CSS result
+ */
+function combineCss(templates, scopedCss) {
 	if (!Array.isArray(scopedCss)) {
 		scopedCss = [scopedCss];
 	}
-	return [...Object.keys(components).map(name => components[name].css), ...scopedCss].join('\n').trim();
+	return [
+		...Object.keys(templates).map(name => templates[name].css),
```
Comment (marionebl) [inline src/css.js]: I'm spoilt by babel-polyfill, I guess 🐛

**Pull request merged**